### PR TITLE
- ANC details screen

### LINKED ISSUE #286: CHW/ANC SubTask - User can view a patient's details
Patient Details 
- [ ] Name , ID
- [ ] Upcoming  service 
- [ ] Last seen  - Encounters 

1.5 days FTE + 20% buffer

image

### PR DESCRIPTION
**Pull request by RaaziaTarique**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #286 

**Type**
Choose one: (Feature) 

**Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [x] I have added any strings visible on UI components to the `strings.xml` file
- [x] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

**Comment by RaaziaTarique** [comment]
@ellykits please review this PR.

File: android/anc/src/main/java/org/smartregister/fhircore/anc/ui/anccare/register/AncItemMapper.kt [hunk 1]
```diff
@@ -31,7 +30,7 @@ object AncItemMapper : DomainMapper<Patient, AncPatientItem> {
 
   override fun mapToDomainModel(dto: Patient): AncPatientItem {
     val name = dto.extractName()
-    val gender = dto.extractGender(AncApplication.getContext()).first()
+    val gender = dto.extractGender().first()
```
Comment (ellykits) [inline android/anc/src/main/java/org/smartregister/fhircore/anc/ui/anccare/register/AncItemMapper.kt:33]: Removing the context will not allow for translation.

File: android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/PatientExtension.kt [hunk 2]
```diff
@@ -36,7 +35,7 @@ private fun String.toTitleCase() = replaceFirstChar {
   if (it.isLowerCase()) it.titlecase(Locale.getDefault()) else it.toString()
 }
 
-fun Patient.extractGender(context: Context) =
+fun Patient.extractGender() =
```
Comment (ellykits) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/PatientExtension.kt:38]: Removing context would not allow for translation. Suggestion is to have the text in `string.xml` file then pick the first letter. Example **Female**  is translated to **Kvenkyns**  in Icelandic (courtesy of Google translate). The first letters are not the same.

**Review by ellykits — CHANGES_REQUESTED** [review]

File: android/anc/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -24,6 +25,7 @@
             </intent-filter>
         </activity>
         <activity android:name=".ui.anccare.register.AncRegisterActivity" />
+        <activity android:name=".ui.anccare.details.AncDetailsActivity" />
```
[the remark this answers is not in the document]
Comment (RaaziaTarique) [reply android/anc/src/main/AndroidManifest.xml:28]: I am the theme at Application level

File: android/anc/src/main/java/org/smartregister/fhircore/anc/ui/anccare/details/CarePlanAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import java.util.Date
+import org.smartregister.fhircore.anc.data.model.CarePlanItem
+import org.smartregister.fhircore.anc.databinding.ItemCareplanBinding
+
+/** Subclass of [ListAdapter] used to display careplan for the ANC client */
+class CarePlanAdapter :
```
[the remark this answers is not in the document]
Comment (RaaziaTarique) [reply android/anc/src/main/java/org/smartregister/fhircore/anc/ui/anccare/details/CarePlanAdapter.kt:29]: I don't see that we'll be needing pagination for this adapter.

File: android/anc/src/main/java/org/smartregister/fhircore/anc/ui/anccare/details/AncPatientItemMapper.kt [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    when (AdministrativeGender.valueOf(this.gender.name)) {
+      AdministrativeGender.MALE -> context.getString(R.string.male)
+      AdministrativeGender.FEMALE -> context.getString(R.string.female)
+      AdministrativeGender.OTHER -> context.getString(R.string.other)
+      AdministrativeGender.UNKNOWN -> context.getString(R.string.unknown)
+      AdministrativeGender.NULL -> ""
+    }
+
+  fun Patient.extractAge(): String {
+    if (!hasBirthDate()) return ""
```
Comment (ellykits) [inline android/anc/src/main/java/org/smartregister/fhircore/anc/ui/anccare/details/AncPatientItemMapper.kt:73]: Can we refactor these implementations as extensions and keep them in the file named `PatientExtension`. Do the refactor too for `PatientItemMapper` in eir application module

**Review by ellykits — CHANGES_REQUESTED** [review]

**Review by ellykits — APPROVED** [review]
:+1: LGTM

**Pull request merged**